pyccel/parser/syntactic.py

```diff
@@ -439,65 +452,65 @@ def _visit_UnitaryOperatorNode(self, stmt):
+        first  = self._visit(stmt.first)
         second = self._visit(stmt.second)
+
         if stmt.value == '+':
-            return Add(first, second)
+            expr = PyccelAdd(first, second)
+            return change_priority(expr)
+
         elif stmt.value == '*':
+            expr = PyccelMul(first, second)
+            return change_priority(expr)
 
-            if isinstance(first, (PythonTuple, Tuple, List)):
-                return Dlist(first, second)
-            return Mul(first, second)
         elif stmt.value == '-':
+            expr = PyccelMinus(first, second)
+            return change_priority(expr)
 
-            if isinstance(stmt.second, BinaryOperatorNode) \
-                and isinstance(second, (Add, Mul)):
-                args = second.args
-                second = second._new_rawargs(-args[0], args[1])
-            else:
-                second = Mul(-1, second)
-            return Add(first, second)
         elif stmt.value == '/':
-            if isinstance(second, Mul) and isinstance(stmt.second,
-                                           BinaryOperatorNode):
-                args = list(second.args)
-                second = Pow(args[0], Integer(-1))
-                second = Mul(second, args[1])
-            else:
-                second = Pow(second, Integer(-1))
-            return Mul(first, second)
+            expr = PyccelDiv(first, second)
+            return change_priority(expr)
 
         elif stmt.value == '**':
+            expr = PyccelPow(first, second)
+            return change_priority(expr)
 
-            return Pow(first, second)
         elif stmt.value == '//':
-
-            if isinstance(second, Mul) and isinstance(stmt.second,
-                                           BinaryOperatorNode):
-                args = second.args
-                second = Pow(args[0], Integer(-1))
-                first =  floor(Mul(first, second), evaluate=False)
-                return Mul(first, args[1])
-            else:
-                second = Pow(second, Integer(-1))
-                return floor(Mul(first, second), evaluate=False)
+            expr = PyccelFloorDiv(first, second)
+            return change_priority(expr)
 
         elif stmt.value == '%':
-            return Mod(first, second)
+            expr = PyccelMod(first, second)
+            return change_priority(expr)
         else:
             msg = 'unknown/unavailable BinaryOperatorNode {node}'
             msg = msg.format(node=type(stmt.value))
             raise PyccelSyntaxError(msg)
 
-
     def _visit_BooleanOperatorNode(self, stmt):
 
         first = self._visit(stmt.first)
         second = self._visit(stmt.second)
+
         if stmt.value == 'and':
-            return And(first, second, evaluate=False)
+            if isinstance(second, PyccelOr):
+                args  = second.args
+                if isinstance(second, PyccelAnd):
+                    return PyccelAnd(first, *args)
```

Comment:
second can't be both a PyccelOr and a PyccelAnd

Reply:
yes i will correct that